Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/WorkspaceServiceImpl.java

```diff
@@ -695,26 +697,125 @@ private void removeGroupLog(Group group, Log log) {
+
+        for (GroupFolder g : getGroupFolders(processId)) {
+            if (g.isHasRead() && g.isHasWrite() && g.isHasOwnership()) {
+                groupAccessTypeMap.put(g.getGroup(), AccessType.OWNER);
+            } else if (g.isHasRead() && g.isHasWrite()) {
+                groupAccessTypeMap.put(g.getGroup(), AccessType.EDITOR);
+            } else if (g.isHasRead()) {
+                groupAccessTypeMap.put(g.getGroup(), AccessType.VIEWER);
+            } else {
+                groupAccessTypeMap.put(g.getGroup(), AccessType.NONE);
+            }
+        }
+
+        return groupAccessTypeMap;
+    }
+
+    public void saveLogACL(Integer logId, String groupRowGuid, AccessType accessType) {
+        switch (accessType) {
+            case OWNER:
+                saveLogPermissions(logId, groupRowGuid, true, true, true);
+                break;
+            case EDITOR:
+                saveLogPermissions(logId, groupRowGuid, true, true, false);
+                break;
+            case VIEWER:
+                saveLogPermissions(logId, groupRowGuid, true, false, false);
+                break;
+            case NONE:
+                break;
+        }
+    }
+
+    public void saveProcessACL(Integer logId, String groupRowGuid, AccessType accessType) {
+        switch (accessType) {
+            case OWNER:
+                saveProcessPermissions(logId, groupRowGuid, true, true, true);
+                break;
+            case EDITOR:
+                saveProcessPermissions(logId, groupRowGuid, true, true, false);
+                break;
+            case VIEWER:
+                saveProcessPermissions(logId, groupRowGuid, true, false, false);
+                break;
+            case NONE:
+                break;
+        }
+    }
+
+    public void saveFolderACL(Integer logId, String groupRowGuid, AccessType accessType) {
+        switch (accessType) {
+            case OWNER:
+                saveFolderPermissions(logId, groupRowGuid, true, true, true);
+                break;
+            case EDITOR:
+                saveFolderPermissions(logId, groupRowGuid, true, true, false);
+                break;
+            case VIEWER:
+                saveFolderPermissions(logId, groupRowGuid, true, false, false);
+                break;
+            case NONE:
+                break;
+        }
+    }
+
+    // Delete Log's access right may lead to logical deleting of user metadata, which need username to fill UpdateBy
+    // field
+    public void deleteLogACL(Integer logId, String groupRowGuid, String username) throws UserNotFoundException {
+
+        removeLogPermissions(logId, groupRowGuid, username);
+    }
+
+    public void deleteProcessACL(Integer processId, String groupRowGuid) {
+
+        removeProcessPermissions(processId, groupRowGuid);
+    }
+
+    public void deleteFolderACL(Integer folderId, String groupRowGuid) {
+
+        removeFolderPermissions(folderId, groupRowGuid);
+    }
```

Comment:
Relating to one comment above, these methods deleteXXXACL with a one-to-one mapping to removeXXXPermissions show that ACL = Permissions. That's why the above comment suggests using a better name for getXXXACL (they should be getXXXAccessType instead), and ACL corresponds to Permissions, so suggest using one consistent name in source code: either ACL or Permission, not both. Once consistent notion and name have been determined, these one-to-one mapping methods will not be needed any more.